### PR TITLE
Allows Htmable content on Descriptions in a Checkboxlist

### PR DESCRIPTION
**Pull request by nessimabadi**
Changes the return hint of the getDescription method in order to handle Htmlable content on a description

- [X] Changes have been thoroughly tested to not break existing functionality.
- [X] New functionality has been documented or existing documentation has been updated to reflect changes.
- [X] Visual changes are explained in the PR description using a screenshot/recording of before and after.

### CODE REVIEW

**Comment by nessimabadi** [comment]
Discussion https://github.com/filamentphp/filament/discussions/8192

**Comment by danharrin** [comment]
Thanks

File: docs-assets/screenshots/images/light/forms/fields/checkbox-list/option-descriptions.jpg
Comment (zepfietje) [inline docs-assets/screenshots/images/light/forms/fields/checkbox-list/option-descriptions.jpg:1]: Any idea why the description for `Laravel` renders in black, @danharrin @nessimabadi?

Comment (danharrin) [reply docs-assets/screenshots/images/light/forms/fields/checkbox-list/option-descriptions.jpg:1]: Fixed on 3.x

Comment (zepfietje) [reply docs-assets/screenshots/images/light/forms/fields/checkbox-list/option-descriptions.jpg:1]: Thanks

**Pull request merged**